frontend/src/components/note/NoteCard.vue

```diff
@@ -0,0 +1,102 @@
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+
+import { Card, Divider } from '@/lib/primevue';
+import { userService } from '@/services';
+import { formatDateShort } from '@/utils/formatters';
+
+import type { Ref } from 'vue';
+import type { Note } from '@/types';
+
+// Props
+type Props = {
+  note: Note;
+  submissionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+// State
+const userName: Ref<string> = ref('');
+
+onMounted(async () => {
+  if (props.note.createdBy) {
+    const res = (await userService.searchUsers({ identityId: [props.note.createdBy] })).data;
```

Comment:
Optional: This could be re-written in then notation to avoid needing to make the entire onMounted callback an async function:
```ts
onMounted(() => {
  if (props.note.createdBy) {
    const res = userService.searchUsers({ identityId: [props.note.createdBy] }).then(response => response.data);
```